prbt_hardware_support/test/unit_tests/unittest_modbus_socket_connection_check.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2018 Pilz GmbH & Co. KG
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <gtest/gtest.h>
+
+#include <numeric>
+#include <modbus/modbus.h>
+
+#include <prbt_hardware_support/libmodbus_client.h>
+#include <prbt_hardware_support/modbus_socket_connection_check.h>
+#include <prbt_hardware_support/pilz_modbus_server_mock.h>
+#include <prbt_hardware_support/pilz_modbus_exceptions.h>
+#include <prbt_hardware_support/client_tests_common.h>
+
+namespace modbus_socket_connection_check_test
+{
+using namespace prbt_hardware_support;
+
+// Each testcase should have its own port in order to avoid conflicts between them
+constexpr unsigned int START_PORT{ 20500 };
+constexpr unsigned int END_PORT{ 20600 };
+static unsigned int ACTIVE_PORT_IDX{ 0 };
+static std::vector<unsigned int> PORTS_FOR_TEST(END_PORT - START_PORT);
+
+constexpr unsigned int DEFAULT_REGISTER_SIZE{ 514 };
+constexpr unsigned int DEFAULT_WRITE_IDX{ 512 };
+
+class ModbusSocketConnectionTest : public testing::Test
+{
+public:
+  static void SetUpTestCase();  // NOLINT
+  void TearDown() override;
+  unsigned int testPort();
```

Comment:
Why is this function `public` and not `protected`.
Why is this function not `static`?